src/scripts/main.js

```diff
@@ -1,3 +1,276 @@
+/* eslint-disable no-unused-vars */
 'use strict';
 
-// write code here
+const thead = document.querySelector('thead');
+const table = document.querySelector('table');
+const tbody = document.querySelector('tbody');
+const arrayTr = document.querySelectorAll('tr');
+const arrayCell = [...arrayTr];
+const arrayPeople = [];
+let directionSort = '';
+
+const showNewPersonInTable = table.insertAdjacentHTML('afterend', `
+<form class = 'new-employee-form'>
+  <label>Name:
+    <input name = "name"
+    type = "text"
+    data-qa = "name"
+    >
+  </label>
+  <label>Position:
+    <input name = "position"
+    type="text"
+    data-qa="position"
+    >
+  </label>
+  <label>Office:
+    <select name = "office" type="text" data-qa="office" required="required">
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    </select>
+  </label>
+  <label>Age:
+    <input name = "age"
+    type="number"
+    data-qa="age"
+    >
+  </label>
+  <label>Salary:
+    <input name = "salary"
+    type="number"
+    data-qa="salary"
+    >
+  </label>
+  <button type = 'submit'>Save to table</button>
+`);
+
+const form = document.querySelector('form');
+
+const addNewPersonInTable = form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const data = new FormData(form);
+  const informNewPerson = Object.fromEntries(data.entries());
+
+  if (informNewPerson.name.length < 4
+    || informNewPerson.age < 18
+    || informNewPerson.age > 90) {
+    pushNotification(100, 10,
+      'ERROR',
+      'It is necessary to fill in all fields'
+      + 'of the form to correct!',
+      'error');
+
+    return;
+  } else {
+    pushNotification(100, 10,
+      'SUCCESS',
+      'I congratulate you!',
+      'success');
+  }
+
+  tbody.insertAdjacentHTML('beforeend', `
+   <tr>
+    <td>${informNewPerson.name}</td>
+    <td>${informNewPerson.position}</td>
+    <td>${informNewPerson.office}</td>
+    <td>${informNewPerson.age}</td>
+    <td>${'$' + divideSalary(informNewPerson.salary)}</td>
+</tr>
+`
+  );
+
+  arrayPeople.push(informNewPerson);
+});
+
+const sortPersonInTable = thead.addEventListener('click', (e) => {
+  const headerName = e.target.innerHTML;
+
+  choiceColumnSort(e, 'Name', headerName);
+  choiceColumnSort(e, 'Position', headerName);
+  choiceColumnSort(e, 'Office', headerName);
+  choiceColumnSort(e, 'Age', headerName);
+  choiceColumnSort(e, 'Salary', headerName);
```

Comment:
you call this function 5 times. it's not about performance. try to refactor this. you can use some conditionals or switch

Reply:
Check all such cases. And try to make it simpler.

Reply:
UPD
form should be reset after submit
I can't edit added rows
salary should be required